modules/common/affinity/affinity.go

```diff
@@ -54,3 +62,78 @@ func DistributePods(
 		},
 	}
 }
+
+// DistributePodsOverride - returns rule to ensure that two replicas of the same selector
+// should not run if possible on the same worker node
+func DistributePodsOverride(
+	selectorKey string,
+	selectorValues []string,
+	topologyKey string,
+	overrides *OverrideSpec,
+) (*corev1.Affinity, error) {
+	defaultAffinity := &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			// This rule ensures that two replicas of the same selector
+			// should not run if possible on the same worker node
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				{
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      selectorKey,
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   selectorValues,
+								},
+							},
+						},
+						// usually corev1.LabelHostname "kubernetes.io/hostname"
+						// https://github.com/kubernetes/api/blob/master/core/v1/well_known_labels.go#L20
+						TopologyKey: topologyKey,
+					},
+					Weight: 100,
+				},
+			},
+		},
+	}
+	// patch the default affinity Object with the data passed as input
+	if overrides != nil {
+		patchedAffinity, err := toCoreAffinity(defaultAffinity, overrides)
+		return patchedAffinity, err
+	}
+	return defaultAffinity, nil
+}
+
+func toCoreAffinity(
```

Comment:
This name isn't intuitive to me. My understanding is the function basically patches one affinity with "overrides" (another affinity) and returns the result. Would using a verb like "merge" be accurate? That would lead me to a name like MergeAffinityOverrides().